test/lib/mocked-balena-api.ts

```diff
@@ -52,6 +52,12 @@ api.get(/\/v6\/device\(uuid=%27([0-9a-f]+)%27\)/, (req, res) =>
 	api.balenaBackend!.getDeviceHandler(req, res, _.noop),
 );
 
+api.get(/\/v6\/device/, (req, res) => {
+	const [, uuid] = /uuid eq '([0-9a-f]+)'/i.exec(req.query['$filter']) ?? [];
```

Comment:
Where does the supervisor make a request like this?

Reply:
https://github.com/balena-io/balena-supervisor/blob/ca8e9f8f6f3773896f614b8aa50e5653dc20dfa8/src/lib/api-helper.ts#L25

Reply:
https://github.com/balena-io/balena-supervisor/blob/ca8e9f8f6f3773896f614b8aa50e5653dc20dfa8/src/lib/api-helper.ts#L84-L96

Reply:
@Page- thinking about this, we shouldn't be fetching the device model even if it is only used to check a key. We should, if this is part of the provisioning flow, do a permissions check on a out-of-model endpoint, like the state endpoint, so that the supervisor isn't tied to an API model version. CC @wrboyce as this might be relevant to your work.